### PR TITLE
DRA non-duplicate functions

### PR DESCRIPTION
**Pull request by cram0**
My first functions that aren't duplicates, learning more about the project 👍🏻

### CODE REVIEW

File: src/ric/1AC60.c [hunk 1]
```diff
@@ -89,7 +89,71 @@ void func_80158814(void) {
...
+        if (g_Player.padTapped & PAD_L2) {
+            if (g_Player.D_80072EFC == 0) {
+                func_801587D0();
+                return true;
+            }
+        }
+        return false;
+    }
+
+    if ((g_Player.D_80072EFC != 0) || (g_Player.padTapped & 1)) {
```
Comment (sonicdcer) [inline src/ric/1AC60.c:107]:
```suggestion
    if ((g_Player.D_80072EFC != 0) || (g_Player.padTapped & PAD_L2)) {

File: src/ric/1AC60.c [hunk 2]
```diff
@@ -89,7 +89,71 @@ void func_80158814(void) {
...
+        if (g_Player.padTapped & PAD_LEFT) {
+            g_Entities->posX.val += 0xFFF00000;
+        }
+        if (g_Player.padTapped & PAD_UP) {
+            PLAYER.posY.val += 0xFFF00000;
+        }
+        var_v0_2 = g_Player.padTapped & PAD_DOWN;
+    }
+
+    if (var_v0_2 != 0) {
```
Comment (sonicdcer) [inline src/ric/1AC60.c:137]: likely this:
```suggestion
}

if (g_Player.padTapped & PAD_DOWN) {

**Comment by cram0** [comment]
The PR contains a function from RIC too so I'll update it and add the RIC functions in another PR

File: src/ric/1AC60.c [hunk 3]
```diff
@@ -89,7 +89,71 @@ void func_80158814(void) {
...
+        PLAYER.posY.val += 0x100000;
+    }
+    if (g_Player.padTapped & PAD_CIRCLE) {
+        PLAYER.animCurFrame -= 1;
```
Comment (sonicdcer) [inline src/ric/1AC60.c:141]: Always try to use ++ and -- when possible, there are very special cases where you will find this doesn't match tho.
```suggestion
PLAYER.animCurFrame--;

File: src/ric/1AC60.c [hunk 4]
```diff
@@ -89,7 +89,71 @@ void func_80158814(void) {
...
+    }
+    if (g_Player.padTapped & PAD_SQUARE) {
+        PLAYER.animCurFrame += 1;
+    }
+
+    if (PLAYER.animCurFrame <= 0) {
+        PLAYER.animCurFrame = 0x1;
```
Comment (sonicdcer) [inline src/ric/1AC60.c:148]: We express animation frames in decimal.
```sugestion
PLAYER.animCurFrame = 1;

File: src/ric/1AC60.c [hunk 5]
```diff
@@ -89,7 +89,71 @@ void func_80158814(void) {
...
+    }
+    if (PLAYER.animCurFrame < 0xD4 == 0) {
+        PLAYER.animCurFrame = 0xD3;
+    }
+    FntPrint("charal:%02x\n", PLAYER.animCurFrame);
+    return true;
+}
+// #endif
```
Comment (sonicdcer) [inline src/ric/1AC60.c:156]: leftover comment

**Review by sonicdcer — COMMENTED** [review]
Congratulations on your success at decompiling these functions! I'm seeing impressive work and progress from your part on understanding the workflow, which i'm very happy with.
I left some minor suggestions and details for you. Keep up the good work!

**Comment by cram0** [comment]
I'll close this PR as all functions have their own PR.

**Pull request closed without merging**